### PR TITLE
[SYCL] Fix __s_long_mad_sat built-in function implementation

### PR DESCRIPTION
**Pull request by maximdimakov**
In the function '__s_long_mad_sat', the second condition had an incorrect return value. It must return a negative value, but only positive value was returned.

### CODE REVIEW

File: sycl/source/detail/builtins_integer.cpp [hunk 1]
```diff
@@ -171,7 +171,7 @@ template <typename T> inline T __s_long_mad_sat(T a, T b, T c) {
   if (!neg_prod && mulhi != 0)
     return d::max_v<T>();
   if (neg_prod && mulhi != -1)
-    return d::max_v<T>(); // essentially some other negative value.
+    return d::min_v<T>(); // essentially some other negative value.
```
Comment (bader) [inline sycl/source/detail/builtins_integer.cpp:174]: Please, add a regression test for this fix.

File: sycl/source/detail/scheduler/commands.hpp [hunk 2]
```diff
@@ -509,8 +509,6 @@ class ExecCGCommand : public Command {
   // memory in DispatchNativeKernel.
   // TODO remove when native kernel support is terminated.
   void releaseCG() {
-    assert(MCommandGroup->getType() == CG::RUN_ON_HOST_INTEL &&
-           "Only 'native kernel' is allowed to release command group");
     MCommandGroup.release();
```
Comment (bader) [inline sycl/source/detail/scheduler/commands.hpp:512]: This doesn't seem to be related to the intrinsic fix. Please, create separate PR for this change.

Comment (maximdimakov) [reply sycl/source/detail/scheduler/commands.hpp]: Removed this change

File: sycl/source/detail/builtins_integer.cpp (hunk 1 above)
Comment (maximdimakov) [reply sycl/source/detail/builtins_integer.cpp:174]: Regression test was added

Comment (maximdimakov) [reply sycl/source/detail/builtins_integer.cpp:174]: The regression test was added

**Comment by s-kanaev** [comment]
@againull friendly ping

**Review by againull — APPROVED** [review]
